ext/background/api.js

```diff
@@ -49,15 +45,23 @@ async function refreshSessions(force) {
     const session = _sessionCache[slug];
     // If session newly initialized or close to expiring, refresh session
     if (
-      !session.session_expires_at ||
-      moment(session.session_expires_at) - moment() < sessionExpiryThreshold
+      !session.session_max_age ||
+      !session.local_session_expires_at ||
+      moment(session.local_session_expires_at) - moment() < sessionExpiryThreshold
     ) {
       const cookieValue = session.cookie_value;
       delete _sessionCache[slug];
       try {
         const newSession = await rangeLogin(slug);
         reportFirstAction(USER_ACTIONS.FIRST_LOGIN, newSession);
         newSession.cookie_value = cookieValue;
+        // Convert max age duration to local time. Cannot rely on session.session_expires_at because
+        // the user's computer might be using the wrong time. This caused a lot of login spam
+        // requests if a user's computer thought that its time was after the session expiration
+        // time.
+        newSession.local_session_expires_at = moment()
+          .add(session.session_max_age, 'milliseconds')
```

Comment:
Reminder that we'll need to update this to seconds to match the server once we make that change.